container/state.go

```diff
@@ -308,8 +305,37 @@ func (s *State) SetRestarting(exitStatus *ExitStatus) {
 	s.FinishedAt = time.Now().UTC()
 	s.ExitCodeValue = exitStatus.ExitCode
 	s.OOMKilled = exitStatus.OOMKilled
-	close(s.waitStop) // fire waiters for stop
-	s.waitStop = make(chan struct{})
+
+	s.notifyWaiters(false)
+}
+
+func (s *State) notifyWaiters(removed bool) {
```

Comment:
Yup, I tried to be quick about it so just decided to go with a shortcut here, since this is only a 'private' function 😄 
Done it in a bit more explicit way.